django/db/models/indexes.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import unicode_literals
+
+import hashlib
+
+from django.utils.encoding import force_bytes
+
+__all__ = ['Index']
+
+# Max length restriction on names imposed by Oracle
+MAX_NAME_LENGTH = 30
```

Comment:
Would be better if this were delegated to the database backend rather than requiring a limit based on one particular database.